src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1501,22 +1514,28 @@ internal void UpdateModelValue(IEnumerable<Guid> modelGuids, string propertyName
 
         internal string GetStringRepOfWorkspace()
         {
-            // Create the xml document to write to.
-            var document = new XmlDocument();
-            document.CreateXmlDeclaration("1.0", null, null);
-            document.AppendChild(document.CreateElement("Workspace"));
-
-            //This is only used for computing relative offsets, it's not actually created
-            string virtualFileName = Path.Combine(Path.GetTempPath(), "DynamoTemp.dyn");
-            Utils.SetDocumentXmlPath(document, virtualFileName);
+            try
+            {
+                //Send the data in JSON format
+                //step 1: convert the model to json
+                var json = this.ToJson(null);
 
-            if (!PopulateXmlDocument(document))
-                return String.Empty;
+                //step2 : parse it as JObject
+                var jo = JObject.Parse(json);
 
-            //Now unset the temp file name again
-            Utils.SetDocumentXmlPath(document, null);
+                //step 2: raise the event to populate the view block
+                if (PopulateJSONWorkspace != null)
```

Comment:
Under what conditions would this be null?

Reply:
I.e. do we need to trap and report this as an error, or are we ok just silently returning only part of the data?

Reply:
In the current scenario, when the VIEW is not tied, this will be null.

Reply:
OK - in which case, can we put a message in the instrumentation saying that there isn't a view, rather than a null?

Reply:
@lukechurch There won't be a view block here. Do you want to add a View block and say there isn't a view?
```
view : { "there is no view attached" }

```

Reply:
I think the lack of a "view" is enough of an indication.